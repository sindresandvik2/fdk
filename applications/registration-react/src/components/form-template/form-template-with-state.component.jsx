import { withState, withHandlers, compose } from 'recompose';
import { FormTemplate } from './form-template.component';

const enhance = compose(
  withState('collapse', 'toggleCollapse', ({ showInitially }) => showInitially),
  withHandlers({
    onToggle: props => () => {
      props.toggleCollapse(!props.collapse);
    }
  })
);

export const FormTemplateWithState = enhance(FormTemplate);
