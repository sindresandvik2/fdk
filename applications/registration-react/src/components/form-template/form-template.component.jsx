import React from 'react';
import PropTypes from 'prop-types';
import cx from 'classnames';
import { Collapse } from 'reactstrap';
import _ from 'lodash';

import localization from '../../utils/localization';
import './form-template.scss';

export const FormTemplate = props => {
  const {
    title,
    backgroundBlue,
    values,
    syncErrors,
    required,
    collapse,
    onToggle
  } = props;

  const collapseClass = cx('fdk-reg_collapse', {
    'fdk-reg_backgroundDefault': !backgroundBlue,
    'fdk-reg_backgroundBlue': backgroundBlue,
    'fdk-reg_collapse_open': collapse
  });

  const buttonClass = cx('fdk-collapseButton', 'fdk-btn-no-border', 'w-100', {
    'p-0': !backgroundBlue
  });

  const collapseIconClass = cx('fa', 'fa-2x', 'mr-2', {
    'fa-angle-down': !collapse,
    'fa-angle-up': collapse
  });

  const collapseContentClass = cx('mt-3', {
    'fdk-collapseContent': backgroundBlue
  });

  return (
    <div className={collapseClass}>
      <button className={buttonClass} onClick={onToggle}>
        <div className="d-flex align-items-center">
          <i className={collapseIconClass} />
          <h2 className="mb-0 text-ellipsis">{title}</h2>
          {required && (
            <span className="fdk-badge badge badge-secondary ml-2 fdk-text-size-small">
              {localization.helptext.required}
            </span>
          )}
          {syncErrors && (
            <div className="d-flex justify-content-end fdk-syncError-icon">
              <i className="fa fa-exclamation-triangle fdk-color-red ml-2" />
            </div>
          )}
        </div>
        {!collapse &&
          values && (
            <div className="d-flex text-left fdk-text-size-small fdk-color3">
              <i className="fa fa-2x fa-angle-down mr-2 visibilityHidden" />
              <span className="text-ellipsis">{values}</span>
            </div>
          )}
      </button>
      <Collapse className={collapseContentClass} isOpen={collapse}>
        {props.children}
      </Collapse>
    </div>
  );
};

FormTemplate.defaultProps = {
  values: null,
  title: null,
  backgroundBlue: false,
  syncErrors: false,
  required: false,
  children: null,
  collapse: false,
  onToggle: _.noop()
};

FormTemplate.propTypes = {
  values: PropTypes.string,
  title: PropTypes.string,
  backgroundBlue: PropTypes.bool,
  syncErrors: PropTypes.oneOfType([
    PropTypes.bool,
    PropTypes.object,
    PropTypes.array
  ]),
  required: PropTypes.bool,
  children: PropTypes.object,
  collapse: PropTypes.bool,
  onToggle: PropTypes.func
};
