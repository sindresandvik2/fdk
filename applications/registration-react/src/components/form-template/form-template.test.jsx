import React from 'react';
import { shallow } from 'enzyme';
import { FormTemplate } from './form-template.component';

let defaultProps;
let wrapper;

beforeEach(() => {
  defaultProps = {
    values: null,
    title: null,
    backgroundBlue: false,
    syncErrors: false,
    required: false,
    children: null
  };
  wrapper = shallow(<FormTemplate {...defaultProps} />);
});

test('should render FormTemplate correctly', () => {
  expect(wrapper).toMatchSnapshot();
});
