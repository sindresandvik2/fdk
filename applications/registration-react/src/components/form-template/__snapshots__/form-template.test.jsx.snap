// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render FormTemplate correctly 1`] = `
<div
  className="fdk-reg_collapse fdk-reg_backgroundDefault"
>
  <button
    className="fdk-collapseButton fdk-btn-no-border w-100 p-0"
  >
    <div
      className="d-flex align-items-center"
    >
      <i
        className="fa fa-2x mr-2 fa-angle-down"
      />
      <h2
        className="mb-0 text-ellipsis"
      />
    </div>
  </button>
  <Collapse
    appear={false}
    className="mt-3"
    enter={true}
    exit={true}
    in={false}
    isOpen={false}
    mountOnEnter={false}
    onEnter={[Function]}
    onEntered={[Function]}
    onEntering={[Function]}
    onExit={[Function]}
    onExited={[Function]}
    onExiting={[Function]}
    tag="div"
    timeout={350}
    unmountOnExit={false}
  />
</div>
`;
